### ISSUE TITLE
Docs updated before images released

### ISSUE THREAD

**Issue by timhughes** [rook/rook#5601]
This is intended as constructive feedback on user experience

I have just encountered a documentation vs release race condition.

I have just had a bit of fun following  the quickstart guide and in particular the TLDR on https://rook.io/docs/rook/v1.3/ceph-quickstart.html

I was luck enough to get the latest and greatest commit https://github.com/rook/rook/commit/06b449a64cd36e8ae8c913a1e4c22604d233916e which updated the image for the operator to `rook/ceph:v1.3.5`

Unfortunately this hadn't managed to make it to the public container registry so I ended up in an ImagePullBackoff situation. 

I have managed to work out what to do thanks to some of your other documentation (thanks for good docs) 

```
kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.3.4
```

You can close this as soon as you read it because it is no longer affecting me. I just thought I would mention this as this was my first experience with Rook and for a while I thought it was just broken. A bit frustrating but on the upside I now know how to switch between minor upgrades. 

Cheers for all the work on this.

**Comment by travisn** [comment]
@timhughes Thanks for the feedback, this is a known issue with no clear solution yet for the timing on updating the release branch. There is typically a delay of ~2 hours between the manifests being updated in the release branch and the release being published. 

One small fix to the documentation that may help is that the latest tag should be recommended instead of the release branch. For example, in the [ceph quickstart](https://rook.github.io/docs/rook/v1.4/ceph-quickstart.html#tldr) document this:
```
git clone --single-branch --branch v1.4.2 https://github.com/rook/rook.git
```
instead of:
```
git clone --single-branch --branch release-1.4 https://github.com/rook/rook.git
```
The docs won't be published until the build is officially published.

**Comment by travisn** [comment]
These links have been fixed for the v1.4.3 and 1.3.11 help docs once published in the next hour or two...
